keras_nlp/src/models/mix_transformer/mix_transformer_backbone.py

```diff
@@ -0,0 +1,157 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import keras
+import numpy as np
+from keras import ops
+
+from keras_nlp.src.api_export import keras_nlp_export
+from keras_nlp.src.models.feature_pyramid_backbone import FeaturePyramidBackbone
+from keras_nlp.src.models.mix_transformer.mix_transformer_layers import (
+    HierarchicalTransformerEncoder,
+)
+from keras_nlp.src.models.mix_transformer.mix_transformer_layers import (
+    OverlappingPatchingAndEmbedding,
+)
+
+
+@keras_nlp_export("keras_nlp.models.MiTBackbone")
+class MiTBackbone(FeaturePyramidBackbone):
+    def __init__(
+        self,
+        depths,
+        include_rescaling=True,
+        input_image_shape=(224, 224, 3),
+        embedding_dims=None,
+        **kwargs,
+    ):
+        """A Backbone implementing the MixTransformer.
+
+        This architecture to be used as a backbone for the SegFormer
+        architecture [SegFormer: Simple and Efficient Design for Semantic
+        Segmentation with Transformers](https://arxiv.org/abs/2105.15203)
+        [Based on the TensorFlow implementation from DeepVision](
+            https://github.com/DavidLandup0/deepvision/tree/main/deepvision/models/classification/mix_transformer)
+
+        Args:
+            depths: the number of transformer encoders to be used per stage in the
+                network.
+            include_rescaling: bool, whether to rescale the inputs. If set
+                to `True`, inputs will be passed through a `Rescaling(1/255.0)`
+                layer. Defaults to `True`.
+            input_image_shape: optional shape tuple, defaults to (224, 224, 3).
+            embedding_dims: the embedding dims per hierarchical stage, used as
+                the levels of the feature pyramid
+
+        Examples:
+
+        Using the class with a `backbone`:
+
+        ```python
+        images = np.ones(shape=(1, 96, 96, 3))
+        labels = np.zeros(shape=(1, 96, 96, 1))
+        backbone = keras_nlp.models.MiTBackbone.from_preset("mit_b0_imagenet")
+
+        # Evaluate model
+        model(images)
+
+        # Train model
+        model.compile(
+            optimizer="adam",
+            loss=keras.losses.BinaryCrossentropy(from_logits=False),
+            metrics=["accuracy"],
+        )
+        model.fit(images, labels, epochs=3)
+        ```
+        """
+        drop_path_rate = 0.1
+        dpr = [x for x in np.linspace(0.0, drop_path_rate, sum(depths))]
+        blockwise_num_heads = [1, 2, 5, 8]
+        blockwise_sr_ratios = [8, 4, 2, 1]
+        num_stages = 4
+
+        # === Layers ===
+        cur = 0
+        patch_embedding_layers = []
+        transformer_blocks = []
+        layer_norms = []
+
+        for i in range(num_stages):
+            patch_embed_layer = OverlappingPatchingAndEmbedding(
+                project_dim=embedding_dims[i],
+                patch_size=7 if i == 0 else 3,
```

Comment:
basically everywhere we see magic numbers like this, consider pulling into config. let's pay attention to other backbones here and try to be consistent.

Where we have common transformer args here try to follow our NLP models. E.g. the number of stack transformer blocks should be `num_layers`, and have no default. For any common vision args check our other vision backbones.

Reply:
@mattdangerw , Shall I add patch_size and stride also to the config?. 
If yes, then does it have to be list of ints for different layers/stages?

Reply:
I would think so. I think probably the most consistent with our current transformer models would be to call `embedding_dims` -> `hidden_dims`. `num_stages` -> `num_layers`.

And then have something like...
```
num_layers=x,
hidden_dims=[...],
patch_sizes=[...],
strides=[...],
```

Unless we want this
```
stackwise_hidden_dims=[...],
stackwise_patch_sizes=[...]
stackwise_strides=[...]
```

Not exactly sure when we should use the "stack" term. Is stack something more specific to convnets?

Reply:
Yes, In the transformer models "stack" term will not make sense. I think we can go with first option.

Reply:
sgtm!